addons/stock/product.py

```diff
@@ -307,27 +323,8 @@ def _product_available(self, cr, uid, ids, name, arg, context=None):
 
     def _search_product_quantity(self, cr, uid, obj, name, domain, context):
         prod = self.pool.get("product.product")
-        res = []
-        for field, operator, value in domain:
-            #to prevent sql injections
-            assert field in ('qty_available', 'virtual_available', 'incoming_qty', 'outgoing_qty'), 'Invalid domain left operand'
-            assert operator in ('<', '>', '=', '!=', '<=', '>='), 'Invalid domain operator'
-            assert isinstance(value, (float, int)), 'Invalid domain right operand'
-
-            if operator == '=':
-                operator = '=='
-
-            product_ids = prod.search(cr, uid, [], context=context)
-            ids = []
-            if product_ids:
-                #TODO: use a query instead of this browse record which is probably making the too much requests, but don't forget
-                #the context that can be set with a location, an owner...
-                for element in prod.browse(cr, uid, product_ids, context=context):
-                    if eval(str(element[field]) + operator + str(value)):
-                        ids.append(element.id)
-            res.append(('product_variant_ids', 'in', ids))
-        return res
-
+        template_ids = prod.search(cr, uid, domain, context=context)
```

Comment:
Technically it is wrong as the qty_available of a product template is the sum of the qty_available of its variants